### PR TITLE
Pop spinup

### PR DESCRIPTION
**Pull request by ipelupessy**
try again

### CODE REVIEW

File: src/omuse/community/pop/Makefile [hunk 1]
```diff
@@ -32,7 +30,7 @@ FS_FLAGS += $(NETCDF_LIBS) $(NETCDFF_LIBS)
 CODE_GENERATOR ?= $(AMUSE_DIR)/build.py
 
 POPREPO=https://github.com/nlesc-smcm/eSalsa-POP
-POPREPOBRANCH=master
+POPREPOBRANCH=spinup_benchmark
```
Comment (ipelupessy) [inline src/omuse/community/pop/Makefile:34]: needs to change whenever we merge spinup_benchmark branch on eSalsa-POP

File: src/omuse/community/pop/interface.f90 [hunk 2]
```diff
@@ -1139,6 +1187,7 @@ function set_element3d_temperature(i, j, k, temp_, n) result (ret)
   real*8, dimension(n), intent(in) :: temp_
 
   call set_gridded_variable_vector_3D(i, j, k, TRACER(:,:,:,1,curtime,:), temp_, n)
+  call set_gridded_variable_vector_3D(i, j, k, TRACER(:,:,:,1,oldtime,:), temp_, n)
```
Comment (ipelupessy) [inline src/omuse/community/pop/interface.f90:1190]: this is better than not setting oldtime, but may be improved (at a price of adding variables (ie the time derivative)), for the moment ok

File: src/omuse/community/pop/interface.py [hunk 3]
```diff
@@ -96,12 +96,23 @@ def set_node_coriolis_f(i=0,j=0,coriolis_f=0. | units.s**-1):
     ##getters for node and element state
     @remote_function(must_handle_array=True)
     def get_node_surface_state(i=0,j=0):
-        returns (ssh=0. | units.cm, vx=0. | units.cm / units.s, vy=0. | units.cm / units.s)
+        returns (ssh=0. | units.cm, vx=0. | units.cm / units.s,     \
+                 vy=0. | units.cm / units.s, gradx=0.  | units.cm**2, \
+                 grady=0. | units.cm**2)
+
```
Comment (ipelupessy) [inline src/omuse/community/pop/interface.py:102]: unit of gradx, grady correct?

File: src/omuse/community/pop/interface.f90 [hunk 4]
```diff
@@ -1811,6 +1864,23 @@ function change_directory(pathname) result (ret)
...
+  character (char_len), intent(in) :: type_
+  ret=0
+  if (type_ == 'file')then
+    shf_data_type = type_
+  elseif(type_ == 'amuse') then
+    shf_data_type = type_
+    shf_formulation = 'restoring' 
+    shf_interp_freq = 'every-timestep'
+    shf_interp_type = 'linear'
+  elseif(type_ == 'amuse') then
```
Comment (ipelupessy) [inline src/omuse/community/pop/interface.f90:1878]: this repeats condition...

**Comment by ipelupessy** [comment]
previous comment still valid:"I think we can have more flexibility if the forcings (shf and sfwf) can be specified through the interface - instead of hard coded analytic expression, same goes for grid..."
but this needs (temporary?) arrays for shf and sfwf to be added ...

**Comment by ipelupessy** [comment]
if the comments are addressed I think it can be merged

File: src/omuse/community/pop/interface.py (hunk 3 above)
Comment (mchertova) [reply src/omuse/community/pop/interface.py:102]: no, it should be units.s**-1

File: src/omuse/community/pop/interface.f90 (hunk 4 above)
Comment (mchertova) [reply src/omuse/community/pop/interface.f90:1878]: Should be combined into elseif(type_=='amuse' .or. type_=='analytic')

**Comment by ipelupessy** [comment]
superseded by #41, but note above comments

**Pull request closed without merging**